libdnf/utils/locker.cpp

```
@@ -0,0 +1,61 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "locker.hpp"
+
+#include <fcntl.h>
+#include <string.h>
+#include <unistd.h>
+
+namespace libdnf::utils {
+
+Locker::LockResult Locker::lock() {
+    lock_fd = open(path.c_str(), O_CREAT | O_RDWR, 0660);
+    if (lock_fd == -1) {
+        return LockResult::ERROR_FD;
```

This just hides any error that occurred during the `open()` call behind an opaque `ERROR_FD` value.